### PR TITLE
(cleanup/scheduler): remove unused function

### PR DESCRIPTION
**Pull request by yuzhiquan**
**What type of PR is this?**

 /kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Review by wgliang — APPROVED** [review]
/lgtm

**Comment by ahg-g** [comment]
This function is used to register out of tree plugins, we shouldn't remove it.

Please update the comment saying: 

"// WithPlugin creates an Option based on plugin name and factory. Please don't remove this function: it is used to register out-of-tree plugins, hence there are no references to it from the kubernetes scheduler code base"

**Comment by yuzhiquan** [comment]
> This function is used to register out of tree plugins, we shouldn't remove it.
> 
> Please update the comment saying:
> 
> "// WithPlugin creates an Option based on plugin name and factory. Please don't remove this function: it is used to register out-of-tree plugins, hence there are no references to it from the kubernetes scheduler code base"

ok, update comment.
 i have used this function before, but can not found more usage now, so send this pr for cleanup.

File: cmd/kube-scheduler/app/server.go [hunk 1]
```diff
@@ -332,9 +332,10 @@ func getRecorderFactory(cc *schedulerserverconfig.CompletedConfig) profile.Recor
...
 
-// WithPlugin creates an Option based on plugin name and factory. This function is used to register out-of-tree plugins.
+// WithPlugin creates an Option based on plugin name and factory. Please don't remove this function: it is used to register out-of-tree plugins,
+// hence there are no references to it from the kubernetes scheduler code base
 func WithPlugin(name string, factory framework.PluginFactory) Option {
 	return func(registry framework.Registry) error {
 		return registry.Register(name, factory)
 	}
-}
+}
```
Comment (ahg-g) [inline cmd/kube-scheduler/app/server.go]: revert this, add empty line at the end

File: cmd/kube-scheduler/app/server.go [hunk 2]
```diff
@@ -332,9 +332,10 @@ func getRecorderFactory(cc *schedulerserverconfig.CompletedConfig) profile.Recor
 	}
 }
 
-// WithPlugin creates an Option based on plugin name and factory. This function is used to register out-of-tree plugins.
+// WithPlugin creates an Option based on plugin name and factory. Please don't remove this function: it is used to register out-of-tree plugins,
+// hence there are no references to it from the kubernetes scheduler code base
```
Comment (ahg-g) [inline cmd/kube-scheduler/app/server.go]: end comment with a "."

**Pull request merged**